### ISSUE TITLE
API for media recording

### ISSUE THREAD

**Issue by nanowang** [flutter/flutter#5736]
This is a feature request so I'll just omit the `repro` stuff.

Both platforms provide APIs like `MediaRecorder` and `AV Foundation` to record audio/video.

That's a major feature for apps involve A/V processing, like IM with voice chatting.

Happy to help implement.

**Comment by eseidelGoogle** [comment]
This should be possible via our hello_services plugin model: https://github.com/flutter/flutter/tree/master/examples/hello_services but I suspect we should provide an example which does this.

**Comment by sethladd** [comment]
Thanks for the feedback!

We don't have plans to address this directly, but we do want to make this possible. In the meantime, please try the hello_services and if you can't get media recording working, please let us know what we can do in the services architecture to make it possible for you.

**Comment by eseidelGoogle** [comment]
https://github.com/flutter/flutter/tree/master/examples/hello_services is the hello_services seth means. :)

**Comment by jadbox** [comment]
It would bemore preferable to have MediaRecorder available natively than to do this service "stitching" between Android Java and and Flutter. I'm rather surprised it's not a bigger concern.

**Comment by eseidelGoogle** [comment]
@jadbox I'm not sure what you mean by "available natively"?

For many applications having sound is of course a big concern! :) However so far the applications we've worked with internally have not needed sound, hence we've done very little work in this area.  We had a very minimal sound API over a year ago when someone built a game on top of Flutter.  But that sound system was built on a different plugin/services architecture than we have now so we deleted it when moving to our new plugin system.  I expect sound is one of many areas we will need to invest in before Flutter is ready for a wider set of applications.
